### PR TITLE
Gohan template improvements

### PR DESCRIPTION
**Pull request by piotr-dulikowski**
- Add schemaMap as a gohan template argument
- Include array properties in GetAllPropertiesFullyQualifiedMap function
- Add filtering of policies by token scope in gohan template command

### CODE REVIEW

File: cli/template.go [hunk 1]
```diff
@@ -363,16 +370,31 @@ func isMatchingPolicy(action schema.Action, policy *schema.Policy) bool {
...
+
+		matchesScope := false
+		for _, tokenType := range scope {
+			if policy.HasTokenType(tokenType) {
+				matchesScope = true
+				break
+			}
+		}
+
+		if !matchesScope {
```
Comment (p-kozlowski) [inline cli/template.go]: let's simplify the control flow a little bit:
```go
func matches(policy, principal, scopes) bool {
  if !matchesPrincipal(policy, principal) {
    return false
  }
  if !matchesScopes(policy, scopes) {
    return false
  }
  return true
}

func filterPolicies(...) {
  ...
  for _, policy := range policies {
    if matches(policy, principal, scopes) {
      append(...)
    }
}
```

File: cli/template.go [hunk 2]
```diff
@@ -363,16 +370,31 @@ func isMatchingPolicy(action schema.Action, policy *schema.Policy) bool {
...
 	var matchedPolicies []*schema.Policy
 	for _, policy := range policies {
-		if policy.Principal == principal {
-			matchedPolicies = append(matchedPolicies, policy)
+		if policy.Principal != principal {
+			continue
 		}
+
+		matchesScope := false
+		for _, tokenType := range scope {
```
Comment (p-kozlowski) [inline cli/template.go]: naming: we're mixing "scope" and "tokenType". I suggest `for _, scope := range scopes`

**Comment by morrisson** [comment]
but need an investigation of circleci failure

**Comment by morrisson** [comment]
Please squash commits before merge

**Pull request merged**